pkg/controller/authorizeec2securitygroupingress/authorizeec2securitygroupingress_controller.go

```diff
@@ -234,6 +245,24 @@ func (r *ReconcileAuthorizeEC2SecurityGroupIngress) Reconcile(request reconcile.
 			return reconcile.Result{}, err
 		}
 		r.events.Event(instance, `Normal`, `Deleted`, "Deleted AuthorizeEC2SecurityGroupIngress and removed finalizers")
+
+		err = r.Get(context.TODO(), types.NamespacedName{Name: instance.Spec.EC2SecurityGroupName, Namespace: instance.Namespace}, ec2SecurityGroup)
+
+		// remove finalizer from parent security group
+		for i, f := range ec2SecurityGroup.ObjectMeta.Finalizers {
```

Comment:
This works for now, but when we refactor I'd prefer to see the ingress rule resource names in a list stored in annotations. Then use a single additional finalizer that simply reads that list.

Reply:
I'll make it happen!